clarify explanation of map_values, including description of new behavior

pkoppstein opened the pull request:
#2466

pkoppstein committed:
manual.yml: clarify explanation of map_values, including description of new behavior

#2466

wader commented on docs/content/manual/manual.yml:
```diff
@@ -925,6 +929,15 @@ sections:
+
+          - program: 'map_values(. // empty)'
+            input: '{"a": null, "b": true, "c": false}'
+            output: ['{"b":true}']
```
Too much to have object with `map` and array with `map_values` examples?

nicowilliams commented on docs/content/manual/manual.yml:
```diff
@@ -908,14 +908,18 @@ sections:
+          `map(f)` is equivalent to `[.[] | f]`. In fact, this is how it's defined.
 
+          `map_values(f)` is similarly defined as `.[] |= f`.  This
+          essentially means that each value, $v, in the input array or
```
`s/, [$]v,//`

nicowilliams commented on docs/content/manual/manual.yml:
```diff
@@ -908,14 +908,18 @@ sections:
+          `map_values(f)` is similarly defined as `.[] |= f`.  This
+          essentially means that each value, $v, in the input array or
+          object is replaced by `first($v|f)`, it being understood
+          that if the input is an object, then keys for which this
```
Array values for which `f` returns `empty` also get deleted.

I liked the original text better.  Staring at the definitions of `map/1` and `map_values/1` the difference between them isn't quite obvious, but the original text here makes it obvious: `map/1` always returns an array, while `map_values/1` returns an array if its input was an array, or an object if its input was an object.

pkoppstein replied:
Tx. Hopefully the subsequent commit addresses all these concerns.  It is more systematic about the various possibilities.

pkoppstein committed:
manual.yml: further tweaks re map and map_values

A more systematic exposition

nicowilliams commented on docs/content/manual/manual.yml:
```diff
@@ -906,15 +906,42 @@ sections:
+          
+          In the absence of errors, `map(f)` always outputs an array
+          whereas `map_values(f)` outputs an array or an object
+          depending on the type of the input.
```
`In the absence of errors, ` is not needed.

Can you add something about the objects' keys being preserved?  Something like:

```
          `map(f)` always outputs an array whereas `map_values(f)` outputs
          an array if given an array, or an object if given an object.  In the
          case of `map_values(f)` applied to an object, the object output
          has all the same keys as the input object minus any for whose
          values `f` produced no new values.
```

something like that.

pkoppstein replied:
Done. Tx.

pkoppstein committed:
manual.yml: further tweaks re map and map_values

Further tweaks, e.g. i.f.o. explicitness:

`map_values(f)` outputs an array if given an array,
or an object if given an object.

nicowilliams commented:
Thanks!
